crates/sui-framework/sources/crypto.move

```diff
@@ -18,6 +18,11 @@ module sui::crypto {
     /// applied to Secp256k1 signatures.
     public native fun ecrecover(signature: vector<u8>, hashed_msg: vector<u8>): vector<u8>;
 
+    /// @param pubkey: A 33-bytes compressed public key, a prefix either 0x02 or 0x03 and a 256-bit integer.
+    /// 
+    /// If the compressed public key is valid, return the 65-bytes uncompressed public key.
+    public native fun decompress_pubkey(pubkey: vector<u8>): vector<u8>;
```

Comment:
(Not to be addressed in this PR, but some ideas for improving this library)
- We could use type wrappers to represent compressed and decompressed public keys, signatures, etc. instead of always using `vector<u8>`. This should make it easier to avoid common mistakes/misuse
- Instead of having one big `crypto` library, we could split it into a few smaller ones partitioned by (e.g.) signature scheme. No strong priors on how this should be done, but I think it will be easier to add new schemes in the future if we can do so by adding a fresh module instead of upgrading an existing on-chain module

Reply:
good point, will follow up a PR: https://github.com/MystenLabs/sui/issues/4567